samples/bluetooth/broadcast_audio_sink/src/main.c

```diff
@@ -83,20 +124,101 @@ static uint8_t sink_broadcast_code[BT_AUDIO_BROADCAST_CODE_SIZE];
 
 uint64_t total_rx_iso_packet_count; /* This value is exposed to test code */
 
+#if defined(CONFIG_USB_DEVICE_AUDIO)
+RING_BUF_DECLARE(ring_buf_usb, AUDIO_RING_BUF_SIZE);
+NET_BUF_POOL_DEFINE(usb_tx_buf_pool, USB_TX_BUF_NUM, BROADCAST_STEREO_SAMPLE_SIZE, 0,
+		    net_buf_destroy);
+
+void mix_mono_to_stereo(void);
+#endif /* defined(CONFIG_USB_DEVICE_AUDIO) */
+
 #if defined(CONFIG_LIBLC3)
+static int16_t audio_buf[MAX_NUM_SAMPLES_MONO];
+static int frames_per_sdu;
+static K_SEM_DEFINE(lc3_decoder_sem, 0, 1);
 
-#include "lc3.h"
+static void do_lc3_decode(struct broadcast_sink_stream *sink_stream);
+static void lc3_decoder_thread(void *arg1, void *arg2, void *arg3);
+K_THREAD_DEFINE(decoder_tid, LC3_ENCODER_STACK_SIZE, lc3_decoder_thread,
+		NULL, NULL, NULL, LC3_ENCODER_PRIORITY, 0, -1);
 
-#define MAX_SAMPLE_RATE       16000
-#define MAX_FRAME_DURATION_US 10000
-#define MAX_NUM_SAMPLES       ((MAX_FRAME_DURATION_US * MAX_SAMPLE_RATE) / USEC_PER_SEC)
+/* Consumer thread of the decoded stream data */
+static void lc3_decoder_thread(void *arg1, void *arg2, void *arg3)
+{
+	while (true) {
+		k_sem_take(&lc3_decoder_sem, K_FOREVER);
+		/* For now we only handle one channel, so always only decode the first */
+		do_lc3_decode(&streams[0]);
+
+#if defined(CONFIG_USB_DEVICE_AUDIO)
+		/* If the ring buffer usage is alrger than zero, then there is data to process */
+		if (ring_buf_space_get(&streams[0].audio_ring_buf)) {
+			mix_mono_to_stereo();
```

Comment:
You claim that we only use 1 channel, but there's nothing in the code below that ensures that.

Reply:
That is correct, this code is the actual guarantee, we will only take data from one channel.

I do want to extend to handle stereo here properly, but for now I would want to keep this PR utilizing only one channel.

Reply:
Right. 

Whether to mix mono to stereo will also depend on what the stream we are synced to does. 

This assumes that mono will always be sent in the stream without checking the codec configuration of that channel, so that should also be ensure before mixing mono, right?

A single stream (BIS) can send stereo as well

Reply:
> I also understand that you could have stereo on a single BIS, and that is basically what I assume right? That the audio is mixed as stereo on a single channel, and then it's duplicated for both USB channels.

Not fully sure. 

If you do stereo on a single BIS, then the left and right channels come after each other (at least for LC3, as it is a mono codec). So if the broadcast source is sending stereo on a single BIS with frame size = 40, then the SDU size is 2 x 40 => 80. 

If you are not doing proper stereo handling in this PR, then I suggest to remove all of it. Having a half-implementation of something in a sample isn't a good solution IMO.

So either only support mono, or properly support stereo.